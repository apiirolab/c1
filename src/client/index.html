<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>