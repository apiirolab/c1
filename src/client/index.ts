export const info = 'Client Info';
