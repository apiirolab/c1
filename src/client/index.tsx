import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
