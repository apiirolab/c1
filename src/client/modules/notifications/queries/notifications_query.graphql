fragment Notifications on Notification {
  id
  createdAt
  text
  type
  processInstance {
    id
    process {
      name
    }
  }
}

query NotificationsQuery($input: NotificationsInput) {
  notificationsQuery(input: $input) {
    ...Notifications
  }
}
