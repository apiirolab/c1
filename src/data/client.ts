import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
