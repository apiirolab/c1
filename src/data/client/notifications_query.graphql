query Notifications {
  notifications {
    id
    date
    text
  }
}
