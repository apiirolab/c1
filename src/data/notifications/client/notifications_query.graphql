query NotificationsQuery($input: NotificationsInput) {
  notifications(input: $input) {
    id
    createdAt
    text
    type
    processInstance {
      id
      process {
        name
      }
    }
  }
}
