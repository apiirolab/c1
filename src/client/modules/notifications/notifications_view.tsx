import * as NOTIFICATIONS_QUERY from 'data/client/notifications_query.graphql';
import * as React from 'react';

import { NotificationsListContainer } from './notification_list_view';
import { NotificationsQuery } from './notifications_queries';

type Props = {
  path?: string;
};

export const NotificationsContainer: React.SFC<Props> = () => (
  <NotificationsQuery query={NOTIFICATIONS_QUERY}>
    {({ loading, error, data }) => {
      if (loading) {
        return 'Loading...';
      }
      if (error) {
        return `Error! ${error.message}`;
      }

      return <NotificationsListContainer notifications={data.notifications} />;
    }}
  </NotificationsQuery>
);
