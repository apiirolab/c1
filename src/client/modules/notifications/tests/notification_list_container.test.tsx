// dog.test.js
import * as QUERY from 'data/notifications/client/notifications_query.graphql';
import * as React from 'react';

import { mock, MockedProvider, QueryTypes, render } from 'client/tests';
import { NotificationsContainer } from '../notifications_view';
import { createNotification } from './notifications_test_data';

export const notificationData = [
  createNotification({ id: '1', createdAt: new Date(2010, 1, 2, 12, 50) }),
  createNotification({ id: '2', createdAt: new Date(2010, 1, 2, 3, 50) }),
  createNotification({ id: '3', createdAt: new Date(2007, 1, 2, 10, 50) }),
  createNotification({ id: '4', createdAt: new Date(2010, 1, 2, 11, 50) }),
  createNotification({ id: '5', createdAt: new Date(2011, 1, 2, 12, 50) })
];

describe('Notifications', () => {
  describe('Container', () => {
    function componentWithData(notifications: QueryTypes.NotificationsNotifications[] = null) {
      if (!notifications) {
        mock.expect(QUERY).reply({
          notifications: [{ id: '1', name: 'one' }, { id: '2', name: 'two' }]
        });
      }

      return (
        <MockedProvider>
          <div>
            <NotificationsContainer />
          </div>
        </MockedProvider>
      );
    }

    it('renders loading', () => {
      mock.expect(QUERY).loading();
      const container = render(componentWithData());

      expect(container).toMatchSnapshot();
      mock.reset();
    });

    it('renders final', async () => {
      mock.expect(QUERY).reply({
        notifications: notificationData
      });

      const container = render(componentWithData());
      expect(container).toMatchSnapshot();
    });
  });
});
