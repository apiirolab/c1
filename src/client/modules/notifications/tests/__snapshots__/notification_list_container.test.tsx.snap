// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Container renders final 1`] = `
<div>
  <h5
    className="ui dividing header"
  >
    <i
      aria-hidden="true"
      className="bell icon"
    />
    <div
      className="content"
    >
      Notifications
    </div>
  </h5>
  <div
    className="ui divided relaxed list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="icon"
      >
        <i
          aria-hidden="true"
          className="grey eye icon sc-bwzfXH kZUMPF"
          onClick={[Function]}
          title="Mark as viewed"
        />
      </i>
      <div
        className="content"
      >
        <div>
          <span
            className="sc-bdVaJa cWCCxy"
          >
            2 months ago
          </span>
           ·
           
          <a
            href="/process/process-name"
            onClick={[Function]}
          >
            Notification Text
          </a>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="icon"
      >
        <i
          aria-hidden="true"
          className="grey eye icon sc-bwzfXH kZUMPF"
          onClick={[Function]}
          title="Mark as viewed"
        />
      </i>
      <div
        className="content"
      >
        <div>
          <span
            className="sc-bdVaJa cWCCxy"
          >
            2 months ago
          </span>
           ·
           
          <a
            href="/process/process-name"
            onClick={[Function]}
          >
            Notification Text
          </a>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="icon"
      >
        <i
          aria-hidden="true"
          className="grey eye icon sc-bwzfXH kZUMPF"
          onClick={[Function]}
          title="Mark as viewed"
        />
      </i>
      <div
        className="content"
      >
        <div>
          <span
            className="sc-bdVaJa cWCCxy"
          >
            2 months ago
          </span>
           ·
           
          <a
            href="/process/process-name"
            onClick={[Function]}
          >
            Notification Text
          </a>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="icon"
      >
        <i
          aria-hidden="true"
          className="grey eye icon sc-bwzfXH kZUMPF"
          onClick={[Function]}
          title="Mark as viewed"
        />
      </i>
      <div
        className="content"
      >
        <div>
          <span
            className="sc-bdVaJa cWCCxy"
          >
            2 months ago
          </span>
           ·
           
          <a
            href="/process/process-name"
            onClick={[Function]}
          >
            Notification Text
          </a>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="icon"
      >
        <i
          aria-hidden="true"
          className="grey eye icon sc-bwzfXH kZUMPF"
          onClick={[Function]}
          title="Mark as viewed"
        />
      </i>
      <div
        className="content"
      >
        <div>
          <span
            className="sc-bdVaJa cWCCxy"
          >
            2 months ago
          </span>
           ·
           
          <a
            href="/process/process-name"
            onClick={[Function]}
          >
            Notification Text
          </a>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ui red icon left labeled button"
    onClick={[Function]}
    role="button"
  >
    <i
      aria-hidden="true"
      className="trash icon"
    />
    Clear All
  </button>
</div>
`;

exports[`Notifications Container renders loading 1`] = `
<div>
  Loading...
</div>
`;
